File: programmers/stack&queue_1.java
//기능개발
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        ArrayList<Integer> arr = new ArrayList<>(); 
        Queue<Integer> qp = new LinkedList<>(); 
        Queue<Integer> qs = new LinkedList<>(); 
        int day = 1;
        for(int i=0;i<progresses.length;i++){
            qp.add(progresses[i]);
        }
        for(int j=0;j<speeds.length;j++){
            qs.add(speeds[j]);
        }
        while(!qp.isEmpty()){
            int cnt = 0;
            while(qp.peek()+(qs.peek()*day)<100){
                day++;
            }
            while(!qp.isEmpty() && qp.peek()+(qs.peek()*day)>=100){
                qp.poll();
                qs.poll();
                cnt++;
            }
            if(cnt>0){
                arr.add(cnt);
            }
            answer = new int[arr.size()];
            for(int i = 0;i<arr.size();i++){
                answer[i] = arr.get(i);
            }
        }

        return answer;
    }
}

File: programmers/stack&queue_1.py
#기능개발
import math
def solution(progresses, speeds):
    answer = []
    queue = []
    for i in range(len(progresses)):
        value = math.ceil((100-progresses[i])/speeds[i]) //올림
        queue.append(value)
        print(value)
    start = queue[0]
    tmp = 0    
    for i in range(len(progresses)):
        if(queue[i]<=start):
            tmp += 1
        else:
            answer.append(tmp)
            start = queue[i]
            tmp = 1
    answer.append(tmp)    
            
        
    return answer
